### PR TITLE
397,398,399,400 Add special events to Pycon site

### LINKED ISSUE #397: Special Event model
New Special Event model with fields for name, slug, location name, start/end dates/times, description (Markdown), and publication status.

### LINKED ISSUE #398: Special events admin editable

### LINKED ISSUE #399: Special event detail page that mimics the CMS style

### LINKED ISSUE #400: Add special events to program export

### PR DESCRIPTION
**Pull request by dpoirier**
* model
* admin (in Django admin)
* detail page that looks like a CMS page
* include in program export

Fixes #397. Fixes #398. Fixes #399. Fixes #400.

### CODE REVIEW

**Comment by rebecca-caktus** [comment]
Merge conflicts here.

**Comment by dpoirier** [comment]
Merge conflicts and a test failure fixed.

**Comment by rebecca-caktus** [comment]
The Special Events model differs very much from the other things in the PyCon admin category. Maybe it deserves its own app?

screenshot

**Comment by rebecca-caktus** [comment]
With that said, the model's verbose name probably won't need to differ from the default.

**Comment by rebecca-caktus** [comment]
Does #400 refer to the schedule export? I thought that [program export](https://github.com/PyCon/pycon/blob/develop/pycon/program_export.py) was intended.

**Comment by dpoirier** [comment]
On Wed, Aug 12, 2015 at 6:20 PM, Rebecca <notifications@github.com> wrote:

> Does #400 <https://github.com/PyCon/pycon/issues/400> refer to the
> schedule export? I thought that program export
> <https://github.com/PyCon/pycon/blob/develop/pycon/program_export.py> was
> intended.
>

​Good question...  maybe it should be both?​

**Comment by dpoirier** [comment]
I'll ask about the exports on today's client call.

**Comment by dpoirier** [comment]
On Wed, Aug 12, 2015 at 6:10 PM, Rebecca <notifications@github.com> wrote:

> The Special Events model differs very much from the other things in the
> PyCon admin category. Maybe it deserves its own app?


​I'm reluctant to add another app (we already have so many), but maybe it
would fit better in a different app than where I've put it?​  Possibly
under Conference?

**Comment by dpoirier** [comment]
Our call yesterday got canceled; I just asked on Basecamp about the exports.

**Comment by dpoirier** [comment]
Ewa says program export (https://basecamp.com/1849796/projects/9507827/messages/46536642#comment_320492231), not schedule export.  I'll update the code.

**Comment by dpoirier** [comment]
@rebecca-caktus I've updated this.

File: pycon/admin.py [hunk 1]
```diff
@@ -136,6 +136,18 @@ class SponsorTutorialAdmin(ProposalMarkEditAdmin):
...
+class SpecialEventAdmin(MarkEditAdmin):
+    list_display = [
+        'name',
+        'start',
+        'end',
+        'location',
+        'published'
+    ]
+    ordering = ['-start']
+    prepopulated_fields = {"slug": ("name",)}
```
Comment (rebecca-caktus) [inline pycon/admin.py]: A couple of nice additions to the admin: a published list filter, and a search box on name and description.

File: pycon/templates/special_event.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{% extends "cms/page_detail.html" %}
+
+{% block box %}
+  <h2>{{ event.location }}</h2>
+  <p>{{ event.start|date }}, {{ event.start|time }} &ndash; {{ event.end|date }}, {{ event.end|time }}</p>
```
Comment (rebecca-caktus) [inline pycon/templates/special_event.html]: Any logic for if the event starts & ends on the same day?

**Comment by rebecca-caktus** [comment]
Changes look good. I've suggested some minor admin customizations & display logic when the event starts & ends on the same day. Then :ship:

**Pull request merged**